### ISSUE TITLE
networkConnections reutrn non-existent  pid on macos 15.1 

### ISSUE THREAD

**Issue by fofolee** [sebhildebrandt/systeminformation#951]
**Describe the bug**
networkConnections reutrn non-existent  pid on macos

macos version is 15.1 （Apple M3 Max）
systeminformation version is 5.23.5

**To Reproduce**
const si = require('systeminformation');
si.networkConnections((x)=>console.log(x))

output:

[
  {
    protocol: 'tcp4',
    localAddress: '192.168.50.235',
    localPort: '59385',
    peerAddress: '**.**.**.**',
    peerPort: '443',
    state: 'ESTABLISHED',
    pid: 131072,
    process: ''
  },
  {
    protocol: 'tcp4',
    localAddress: '192.168.50.235',
    localPort: '59371',
    peerAddress: '    peerAddress: '**.**.**.**',',
    peerPort: '443',
    state: 'ESTABLISHED',
    pid: 131072,
    process: ''
  }
…………
]

ps -ef | grep 131072

output:
nothing

all pids do not exist
In fact, the pid of my process is basically 5 digits, but the pid returned by networkConnections is 6 digits

**Comment by sebhildebrandt** [comment]
@fofolee hi, I just tested it here and I cannot reproduce it. So can you please pass me the output of the following:

```
netstat -natvln | grep "tcp4\\|tcp6\\|udp4\\|udp6"
```

... as text, just a few lines WITH status and some without.

Thank you in advance!

**Comment by fofolee** [comment]
like this?
```
netstat -natvln | grep "tcp4\\|tcp6\\|udp4\\|udp6" | head -n 10
tcp4       0      0  192.168.1.233.64801                           185.199.111.154.443                           ESTABLISHED        958       2052  131147  131600   2646      0 00102 00000008 00000000015a5091 20000081 04000900      2      0 000004
tcp4       0      0  192.168.1.233.64800                           20.205.243.166.443                            ESTABLISHED       7406       1927  131072  131600   2646      0 00102 00000008 00000000015a5090 20000081 04000900      2      0 000004
tcp6       0      0  2408:8244:410:17ac:4ca5:b649:5d67:7504.64799  2603:1040:5:8::2.443                          ESTABLISHED      14222       2974  262144  132240   2646      0 00102 00000008 00000000015a5069 20000081 04000800      2      0 000004
tcp6       0      0  2408:8244:410:17ac:4ca5:b649:5d67:7504.64798  2620:1ec:c11::239.443                         ESTABLISHED      21178       6354  262144  131072   2646      0 00102 00000008 00000000015a5050 20000081 04000800      2      0 000004
tcp4       0      0  192.168.1.233.64797                           20.187.186.89.443                             ESTABLISHED      13294       5379  131072  131072   2646      0 00102 00000008 00000000015a5044 20000081 04000900      2      0 000004
tcp4       0      0  127.0.0.1.9197                                127.0.0.1.64796                               ESTABLISHED        774        150  407965  146988  58194      0 00002 00000004 00000000015a47e2 00000081 01000900      2      0 000000
tcp4       0      0  127.0.0.1.64796                               127.0.0.1.9197                                ESTABLISHED        404        335  408150  146988  58194      0 00102 00020000 00000000015a47e1 00180081 04080900      1      0 000000
tcp4       0      0  10.8.0.2.64794                                13.69.239.72.443                              FIN_WAIT_1       14434       4282  131072  131072  10024      0 0013b 00000008 00000000015a4666 00000001 04000900      0      0 000000
tcp4       0      0  10.8.0.2.64792                                20.205.69.80.443                              FIN_WAIT_1       13726       3428  131072  131908  10024      0 0013b 00000008 00000000015a463e 00000001 04000900      0      0 000000
tcp4       0      0  10.8.0.2.64780                                140.82.112.26.443                             ESTABLISHED       9106       4403  131072  131908   2646      0 00102 00000008 00000000015a41d7 20000001 04000900      2      0 000004
```

i also try to search 64801(which in the first line of netstat above) in the result of systeminformation
```
> si.networkConnections((x)=>console.log(x.filter(y=>y.localPort==='64801')))
> [
  {
    protocol: 'tcp4',
    localAddress: '192.168.1.233',
    localPort: '64801',
    peerAddress: '185.199.111.154',
    peerPort: '443',
    state: 'ESTABLISHED',
    pid: 131147,
    process: ''
  }
]
```

it return pid 131147, but the right pid in netstat is 2646

It seems i miss the problem when i upgrade my macos to 15.1 ,  my macos version is 13 before

**Comment by sebhildebrandt** [comment]
@fofolee thank you for providing the data! For some reason it takes the wrong column on your machine ... I will mock my code with your data to see if I can spot the problem.

**Comment by sebhildebrandt** [comment]
@fofolee Ahhh ... I think I see the reason. Will provide a fix the next couple of days.

**Comment by sebhildebrandt** [comment]
@fofolee Version 5.23.11 just released, can you try it once again on your side?

**Comment by fofolee** [comment]
Thanks for the upgrade
but it still seems to get the wrong column

```
netstat -natvln | grep "tcp4\\|tcp6\\|udp4\\|udp6" | head -n 10
tcp4       0    126  192.168.1.233.52137                           34.117.41.85.443                              ESTABLISHED       9026        316  131072  131600   2903      0 00102 00000000 00000000023c645f 20000081 04000900      2      0 000004
tcp4       0      0  192.168.1.233.52126                           123.57.223.58.443                             ESTABLISHED      17368       1612  262144  131316   5767      0 00102 00000108 00000000023c629c 20000080 04006900      2      0 000004
tcp4       0      0  192.168.1.233.52125                           123.57.223.58.443                             ESTABLISHED      17330       1612  262144  131316  75236      0 00102 00000108 00000000023c629b 20000080 04000900      2      0 000004
tcp4       0      0  192.168.1.233.52094                           47.106.66.231.443                             ESTABLISHED       2480       1197  131072  131600  74164      0 00102 00000100 00000000023c478b 20000081 04000900      2      0 000004
tcp4       0      0  192.168.1.233.52038                           119.167.243.61.443                            ESTABLISHED    5475898       1010  940272  131600  74167      0 00102 00000008 00000000023c454f 20000081 04000900      2      0 000004
tcp6       0      0  2408:8244:410:17ac:2013:52c:ad61:4882.51973   2600:140b:a00:a::b81b:b9de.443                ESTABLISHED       2162       3293  131072  131868   1099      0 00102 00000008 00000000023c3bb5 20000081 04000800      2      0 000004
tcp4       0      0  192.168.1.233.51933                           185.199.109.154.443                           ESTABLISHED     700176       2939  250072  131072   1099      0 00102 00000008 00000000023c36ef 20000081 04000900      2      0 000004
tcp4       0      0  192.168.1.233.51928                           140.82.113.25.443                             ESTABLISHED      10584       4637  131072  131072   1099      0 00102 00000008 00000000023c36b0 20000081 04000900      2      0 000004
tcp4       0      0  192.168.1.233.51927                           185.199.108.154.443                           ESTABLISHED     152072       2949  161872  131072   1099      0 00102 00000008 00000000023c369f 20000081 04000900      2      0 000004
tcp6       0      0  2408:8244:410:17ac:2013:52c:ad61:4882.51926   2606:50c0:8001::154.443                       ESTABLISHED      17368       7930  131072  131072   1099      0 00102 00000008 00000000023c369b 20000081 04000800      2      0 000004
```

right pid is 2903
```
si.networkConnections((x)=>console.log(x.filter(y=>y.localPort==='52137')))
> [
  {
    protocol: 'tcp4',
    localAddress: '192.168.1.233',
    localPort: '52137',
    peerAddress: '34.117.41.85',
    peerPort: '443',
    state: 'ESTABLISHED',
    pid: 131072,
    process: ''
  }
]
```

```
cat package.json 
{
  "dependencies": {
    "systeminformation": "^5.23.11"
  }
}
```

**Comment by sebhildebrandt** [comment]
@fofolee OK, can you provide the following information:

```
netstat -natvln | head -n 10
```

to see the exact header ... seems that there was a change as on my machine (M1 - 14.6.1) the output is correct.

**Comment by sebhildebrandt** [comment]
@fofolee ... header have changed in 15.x ... so we 2 have additional columns (`rxbytes` and `txbytes`) ... I provided a fix for that now. Version 5.23.12 just released. Can you try this latest version?

**Comment by fofolee** [comment]
@sebhildebrandt  
Great, the problem was fixed on 5.23.12, thank you very much.

**Comment by sebhildebrandt** [comment]
@fofolee Thank you for testing. Glad to hear, that this now works!
